drivers/watchdog/wdt_opentitan.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2023 by Rivos Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <soc.h>
+#include <zephyr/drivers/watchdog.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(wdt_opentitan, CONFIG_WDT_LOG_LEVEL);
+
+#define DT_DRV_COMPAT lowrisc_opentitan_aontimer
+
+#define REG_OT_WDT(dev) \
+	(((const struct wdt_ot_aontimer_cfg *const)(dev)->config)->regs)
+#define NODE_AON DT_NODELABEL(aontimer)
+
+struct ot_aontimer_regs {
+	uint32_t alert_test;
+	uint32_t wkup_ctrl;
+	uint32_t wkup_thold;
+	uint32_t wkup_count;
+	uint32_t wdog_regwen;
+	uint32_t wdog_ctrl;
+	uint32_t wdog_bark_thold;
+	uint32_t wdog_bite_thold;
+	uint32_t wdog_count;
+	uint32_t intr_state;
+} __packed;
```

Comment:
why not use offset definitions?

Reply:
I believe this definition is more clear than using offsets.

Reply:
Is it though? Like, can you tell straight away what is the offset of the `wdog_count` register?